sdk/state/ingest_test.go

```diff
@@ -820,8 +821,44 @@ func TestChannel_IngestTx_updateState_invalid_initiatorEscrowHasExtraSigner(t *t
+	validResultXDR, err := txbuildtest.BuildResultXDR(true)
+	require.NoError(t, err)
+	resultMetaXDR, err := txbuildtest.BuildResultMetaXDR(xdr.LedgerEntryChanges{
+		xdr.LedgerEntryChange{
+			Type: xdr.LedgerEntryChangeTypeLedgerEntryUpdated,
+			Updated: &xdr.LedgerEntry{
+				Data: xdr.LedgerEntryData{
+					Type: xdr.LedgerEntryTypeAccount,
+					Account: &xdr.AccountEntry{
+						AccountId: xdr.MustAddress(initiatorEscrow.Address()),
+						SeqNum:    102,
+						Signers: []xdr.Signer{
+							{
+								Key: xdr.SignerKey{
+									Type:    xdr.SignerKeyTypeSignerKeyTypeEd25519,
+									Ed25519: xdr.MustAddress("GAKDNXUGEIRGESAXOPUHU4GOWLVYGQFJVHQOGFXKBXDGZ7AKMPPSDDPV").Ed25519,
+								},
+								Weight: 1,
+							},
+						},
+						Thresholds: xdr.Thresholds{0, 2, 2, 2},
+					},
+				},
+			},
+		},
+		xdr.LedgerEntryChange{
+			Type: xdr.LedgerEntryChangeTypeLedgerEntryUpdated,
+			Updated: &xdr.LedgerEntry{
+				Data: xdr.LedgerEntryData{
+					Type: xdr.LedgerEntryTypeAccount,
+					Account: &xdr.AccountEntry{
+						AccountId: xdr.MustAddress(responderEscrow.Address()),
+					},
+				},
+			},
+		},
+	})
```

Comment:
💡 I think we might be able to get away with only supporting `...interface{}`, and the function can assume it's the after state we're wanting to set and pass the value into `xdr.NewLedgerEntryData`. It would reduce the test code down to:
```suggestion
	resultMetaXDR, err := txbuildtest.BuildResultMetaXDR(
		&xdr.AccountEntry{
			AccountId: xdr.MustAddress(initiatorEscrow.Address()),
			SeqNum:    102,
			Signers: []xdr.Signer{
				{ Key: xdr.MustSigner("GAKDNXUGEIRGESAXOPUHU4GOWLVYGQFJVHQOGFXKBXDGZ7AKMPPSDDPV"), Weight: 1 },
				},
				Thresholds: xdr.Thresholds{0, 2, 2, 2},
			},
		},
		&xdr.AccountEntry{
			AccountId: xdr.MustAddress(responderEscrow.Address()),
		},
	})
```

Reply:
I couldn't think of a good way of fitting `xdr. NewLedgerEntryData` into it, since we may need to pass in multiple ledgerEntryData object of different types (like in `TestChannel_IngestTx_updateState_invalid_initiatorEscrowHasExtraSigner`)

but I changed it to assume it's an update change so we only have to pass in `[]xdr.LedgerEntryData` now, which also really simplifies what we have to pass in.

going to merge since approved but if you see something that can be improved lmk

Reply:
If you change the input type to `entries ...interface{}`, and then inside the function: 
```go
for _, e := range entries {
	switch le := e.(type) {
	case xdr.AccountEntry:
		xdr.NewLedgerEntryData("Account", le)
	case xdr.TrustLineEntry:
		xdr.NewLedgerEntryData("TrustLine", le)
	default:
		panic("unrecognized entry type")
	}
}
```

Reply:
Assuming it's an update sounds good to me.

Reply:
Since we're requiring it to be a `[]xdr.LedgerEntryData{}` anyways any reason to make the input an interface? The input would be the same. Imo I think it's clearer making the inputs to be a required type 🤔

Reply:
I was only suggesting so that we don't have to build the ledger entry data manually in tests. In tests I don't think it matters much if we have interface{}'s. Either way, this is great as is.